docs/core/runtime-discovery/troubleshoot-app-launch.md

```diff
@@ -0,0 +1,157 @@
+
+#### Run an installer
+
+If your existing installation of .NET was installed through an installer or package manager, then also installing the required framework through an installer or package manager is likely the simpler option. Otherwise, you can [download binaries](#download-binaries) instead of using an installer.
+
+In most cases, when the application that failed to launch is using such an installation, the ".NET location" in the error message would point to:
+::: zone pivot="os-windows"
+`%ProgramFiles%\dotnet`
+::: zone-end
+::: zone pivot="os-linux"
+`/usr/share/dotnet/`
+::: zone-end
+::: zone pivot="os-macos"
+`/usr/local/share/dotnet/`
+::: zone-end
+
+::: zone pivot="os-windows,os-macos"
+From the **Installers** column of the runtime download, download the installer matching the required architecture. Run the downloaded installer.
+::: zone-end
+::: zone pivot="os-linux"
+Different Linux distributions provide .NET through different package managers. See [Install .NET on Linux](../install/linux.md) for details. Note that preview versions of .NET are not available through package managers.
+::: zone-end
+
+#### Download binaries
+
+From the **Binaries** column of the runtime download, download the binary release matching the required architecture. Extract the downloaded archive to the ".NET location" specified in the error message.
+
+::: zone pivot="os-windows"
+For more details on manual installation, see [Install .NET on Windows](../install/windows.md#download-and-manually-install)
+::: zone-end
+
+::: zone pivot="os-linux"
+For more details on manual installation, see [Install .NET on Linux](../install/linux.md#manual-installation)
+::: zone-end
+
+::: zone pivot="os-macos"
+For more details on manual installation, see [Install .NET on macOS](../install/macos.md#download-and-manually-install)
+::: zone-end
+
+### Configure roll-forward behavior
+
+If you already have a higher version of the required framework installed, you can make the application ron on that higher version by configuring its roll-forward behavior.
+
+When running the application, you can specify the [`--roll-forward` command line option](../tools/dotnet.md#runtime-options) or setting the [`DOTNET_ROLL_FORWARD` environment variable](../tools/dotnet-environment-variables.md#dotnet_roll_forward).
+By default, an application will require a framework matching the same major version that the application targets, but can use a higher minor or patch version. However, application developers may have specified a different behavior. For more details, see [Framework-dependent apps roll-forward](../versions/selection.md#framework-dependent-apps-roll-forward).
+
+Note that, since using this option can let the application run on a different framework version than the one for which it was designed, it may result in unintended behavior in the application due to changes between versions of a framework.
```

Comment:
```suggestion
> [!NOTE]
> Since using this option lets the application run on a different framework version than the one for which it was designed, it may result in unintended behavior due to changes between versions of a framework.
```